### PR TITLE
refactor: items interpretation

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds the refactoring of the interpretation for `items`.

It changes the entire perspective, to instead of adding simplification functions per CommonModel property, we process the JSON Schema keywords which then affects the CommonModel in some way.

**It adds**
- InterpretationItems, to interpret the `items` schema keyword.

**Files this PR is supposed to replace**
- https://github.com/asyncapi/generator-model-sdk/blob/master/src/simplification/SimplifyItems.ts
- And all related tests have been refactored to have next to no dependencies by using the jest mocking framework.

**Todo**

- [x] Refactor docs
- [x] Solve code smells
- [x] Add logging to the process
- [x] Inspect tests
- [x] Rename to transformer

**Related issue(s)**
Blocked by #184

### CODE REVIEW

File: src/models/CommonModel.ts [hunk 1]
```diff
@@ -82,6 +82,23 @@ export class CommonModel extends CommonSchema<CommonModel> {
...
+   * 
+   * If items already exist the two are merged.
+   * 
+   * @param itemModel 
+   * @param schema schema to the corresponding property model
+   */
+  addItem(itemModel: CommonModel, schema: Schema) {
+    if (this.items !== undefined) {
+      Logger.warn(`While trying to add item to model ${this.$id}, duplicate items found. Merging models together to form a unified item model.`, itemModel, schema, this);
+      this.items = CommonModel.mergeCommonModels(this.items as CommonModel, itemModel, schema);
```
Comment (magicmatatjahu) [inline src/models/CommonModel.ts:96]: Only reminder, we can have tuple type 😄

**Review by magicmatatjahu — DISMISSED** [review]

File: src/models/CommonModel.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/models/CommonModel.ts:96]: Yep, on the milestone list will be changed afterwards - #67

File: src/interpreter/Interpreter.ts [hunk 2]
```diff
@@ -78,7 +79,8 @@ export class Interpreter {
       }
 
       model.required = schema.required || model.required;
-
+      
+      interpretItems(schema, model, this);
```
Comment (magicmatatjahu) [inline src/interpreter/Interpreter.ts:83]: Sorry 😅 I didn't check that

**Comment by jonaslagoni** [comment]
Solved merge conflict and added missing interpreter integration 🤦

**Review by magicmatatjahu — APPROVED** [review]

File: src/interpreter/Interpreter.ts (hunk 2 above)
Comment (jonaslagoni) [reply src/interpreter/Interpreter.ts:83]: No worries, my mistake for not adding it 😄

**Comment by jonaslagoni** [comment]
Thanks!

**Pull request merged**